### PR TITLE
Fix deduplication in datagen

### PR DESCRIPTION
**Pull request by sffc**
Fixes https://github.com/unicode-org/icu4x/issues/4232

I want to put this on the 1.3 branch.

This was discovered by @sven-oly's conformance test suite.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,20 @@
 # Changelog
 
+## icu4x 1.3.3 (later Oct, 2023)
+
+Some crates received a 1.3.3 patch release:
+
+- [Fix #4165, locale "vi" not working in Collator](https://github.com/unicode-org/icu4x/issues/4165):
+    - `icu_collator`
+    - `icu_collator_data`
+- [Fix #4232, some locale data being improperly deduplicated](https://github.com/unicode-org/icu4x/issues/4232):
```
Comment (robertbastian) [inline CHANGELOG.md:10]: can we be more specific? what data exactly did this affect?

**Review by robertbastian — COMMENTED** [review]
There is no testdata that would have caught this. Should we add some more complex locales to testdata?

**Review by Manishearth — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (sffc) [reply CHANGELOG.md:10]: People can click the bug link for more details

Comment (robertbastian) [reply CHANGELOG.md:10]: not clear from the bug for me. It mentions es-MX, but the diff includes other languages as well

**Comment by sffc** [comment]
I merged this via command line in order to keep the commits separate.

File: CHANGELOG.md (hunk 1 above)
Comment (sffc) [reply CHANGELOG.md]: I added a bit more info in the bug.

**Pull request closed without merging**